compiler/rustc_hir_analysis/src/astconv/bounds.rs

```diff
@@ -302,18 +292,11 @@ impl<'tcx> dyn AstConv<'tcx> + '_ {
 
         // We have already adjusted the item name above, so compare with `ident.normalize_to_macros_2_0()` instead
         // of calling `filter_by_name_and_kind`.
-        let find_item_of_kind = |kind| {
-            tcx.associated_items(candidate.def_id())
-                .filter_by_name_unhygienic(assoc_ident.name)
-                .find(|i| i.kind == kind && i.ident(tcx).normalize_to_macros_2_0() == assoc_ident)
-        };
-        let assoc_item = if return_type_notation {
-            find_item_of_kind(ty::AssocKind::Fn)
-        } else {
-            find_item_of_kind(ty::AssocKind::Type)
-                .or_else(|| find_item_of_kind(ty::AssocKind::Const))
-        }
-        .expect("missing associated type");
+        let assoc_item = tcx
+            .associated_items(candidate.def_id())
+            .filter_by_name_unhygienic(assoc_ident.name)
+            .find(|i| i.kind == assoc_kind && i.ident(tcx).normalize_to_macros_2_0() == assoc_ident)
```

Comment:
`find_by_name_and_kind` is semantically identical to the code above but this version is slightly faster in theory since above we only normalize the LHS instead of normalizing both sides. `assoc_ident` has already been normalized above in `adjust_ident_and_get_scope`, so no need to do it again (`adjust_ident_and_get_scope` also returns the `def_scope` which is used later in `is_accessible_from`).

Reply:
pls note this